### ISSUE TITLE
Crossentropy loss and sigmoid output

### ISSUE THREAD

**Issue by oysteijo** [oysteijo/simd_neuralnet#1]
Comparing to Keras, it looks like the gradient calculated with "crossentropy" loss function and "sigmoid" output layer activation functions is off by a scale of n-number of outputs. I believe that the derivative of the loss function should take the mean of the outputs.

There are several ways to solve this. We can either make a special case `sigmoid_crossentropy_derviative()`, or we can make a special `sigmoid_crossentropy_loss()`. Not sure what I prefer.

**Comment by oysteijo** [comment]
This is now fixed by splitting `crossentropy` into `categorical_crossentropy` and `binary_crossentropy`.

`categorical_crossentropy` matches `softmax`
`binary_crossentropy` matches `sigmoid`

This is hence solved by commit d98bd2b
